docs/beginners-guide-python.md

```diff
@@ -0,0 +1,219 @@
+that allows for programmatic interaction via Scala, Python, and Java. SystemML also features an
+embedded API for scoring models.
+
+
+## Automatic Optimization
+
+Algorithms specified in DML are dynamically compiled and optimized based on data and cluster characteristics
+using rule-based and cost-based optimization techniques. The optimizer automatically generates hybrid runtime
+execution plans ranging from in-memory, single-node execution, to distributed computations on Spark or Hadoop.
+This ensures both efficiency and scalability. Automatic optimization reduces or eliminates the need to hand-tune
+distributed runtime execution plans and system configurations.
+
+## ML Algorithms
+
+SystemML features a suite of production-level examples that can be grouped into six broad categories:
+Descriptive Statistics, Classification, Clustering, Regression, Matrix Factorization, and Survival Analysis.
+Detailed descriptions of these algorithms can be found in the
+[SystemML Algorithms Reference](http://apache.github.io/incubator-systemml/algorithms-reference.html).  The goal of these provided algorithms is to serve as production-level examples that can modified or used as inspiration for a new custom algorithm.
+
+## Download & Setup
+
+Before you get started on SystemML, make sure that your environment is set up and ready to go.
+
+  1. **If you’re on OS X, we recommend installing [Homebrew](http://brew.sh) if you haven’t already.  For Linux users, the [Linuxbrew project](http://linuxbrew.sh/) is equivalent.**
+
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install)"
+</div>
+</div>
+  
+  2. **Install Java (need Java 8).**
+ 
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+brew tap caskroom/cask
+brew install Caskroom/cask/java
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+brew tap caskroom/cask
+brew install Caskroom/cask/java
+</div>
+</div>
+
+  3. **Install Spark.**
+
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+brew install apache-spark
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+brew install apache-spark
+</div>
+</div>
+  
+
+  4. **Download SystemML.**
+
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+pip install SystemML
+curl "https://sparktc.ibmcloud.com/repo/latest/SystemML.jar" -o "SystemML.jar"
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+pip install SystemML
+wget https://sparktc.ibmcloud.com/repo/latest/SystemML.jar
+</div>
+</div>
+
+**Congrats! You can now use SystemML!**
+
+## Invoke SystemML's algorithms
+
+{% highlight python %}
```

Comment:
You can wrap the code around with triple backticks and it will show syntax highlighting in the README as well. (See: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting)